src/main/scala/strawman/collection/mutable/Seq.scala

```diff
@@ -0,0 +1,76 @@
+package strawman
+package collection
+package mutable
+
+trait Seq[A] extends strawman.collection.Seq[A] with Growable[A] {
```

Comment:
The problem with having `+=` and other Growable operations produce a new collection on `WrappedArray` is that your `inPlace` operations no longer operate on the collection you wrapped, and it's not at all obvious that this is what happened, since you might go back to the original array and expect to find the changes there.  One could also envision a mutable proxied Seq backed by whatever was useful (not just `Array`)--the equivalent of a lens for a set of mutable values.  We needn't necessarily support these sorts of things, but I think the cost would be relatively low, just like having a distinction between `Seq` and `IndexedOptimizedSeq` is (hopefully) low.